tests/test_cascade.py

```
@@ -92,6 +93,58 @@ def test_margin_threshold_can_enable_a_numeric_large_margin_resolution(self, tmp
         assert result["tier"] == "resolved"
         assert result["reason"] == "heading_nav_tfidf_agree_large_margin"
 
+    #: Non-finite/non-positive numeric values, plus a non-numeric string and
+    #: a bool -- the latter two exist to pin down a real bug (studio#135's
+    #: review): math.isfinite() raises TypeError for either, which would
+    #: propagate out before the intended ValueError ever ran, defeating
+    #: _validate_margin_threshold's documented contract for exactly the
+    #: direct-Python-caller case its docstring exists for. bool is checked
+    #: separately from the numeric cases since it subclasses int and would
+    #: otherwise slip past a bare isinstance(x, (int, float)) check.
+    _BAD_MARGIN_THRESHOLDS = [0, -1, float("nan"), float("inf"), "0.5", True]
```

Ruff (0.16.3)

[warning] Mutable default value for class attribute

(RUF012)